packages/sources/dar/src/transport/price.ts

```diff
@@ -80,4 +92,29 @@ export const transport = new WebSocketTransport<WsTransportTypes>({
       ]
     },
   },
-})
+}
+
+export class DarWebsocketTransport extends WebSocketTransport<WsTransportTypes> {
+  override async streamHandler(
+    context: EndpointContext<WsTransportTypes>,
+    subscriptions: SubscriptionDeltas<RequestParams>,
+  ): Promise<void> {
+    if (
+      this.wsConnection &&
+      !this.connectionClosed() &&
+      (subscriptions.new.length || subscriptions.stale.length)
+    ) {
+      logger.debug(
+        `closing WS connection for new subscriptions: ${JSON.stringify(subscriptions.desired)}`,
+      )
+      const closed = new Promise<void>((resolve) => (this.wsConnection.onclose = resolve))
+      this.wsConnection.close()
+      await closed
+    }
+
+    subscriptions.new = subscriptions.desired
```

Comment:
I think we also need to set `subscriptions.stale = []`, else they will continue to sit in the `stale` list. 

The `TwoSigma` EA does this today, which could be a helpful example: https://github.com/smartcontractkit/external-adapters-js/blob/d21ac3bb9790939a817c17279a751335d72db09b/packages/sources/twosigma/src/transport/price.ts#L110-L132